File: leetcode/contains_duplicate/contains_duplicate.cpp
/*
 * Time Complexity: O(N)
 *   - Each element is inserted and checked in the set once.
 *
 * Space Complexity: O(N)
 *   - In the worst case, all elements are unique and stored in the set.
 */

#include<unordered_set>
#include<vector>

using std::unordered_set;
using std::vector;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> dist;
        for(int num : nums) {
            if(dist.find(num) != dist.end()) return true;
            dist.insert(num);
        }
        return false;
    }
};

File: leetcode/contains_duplicate/contains_duplicate.java
package contains_duplicate;

import java.util.HashSet;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> dist = new HashSet<>();
        for(int num : nums) {
            if(dist.contains(num)) return true;
            dist.add(num);
        }
        return false;
    }
}
